### PR TITLE
Fix bigquery case sensitive caching issue (#1810)

### LINKED ISSUE #1810: Make BigQuery cache lookups case-insensitive
UPDATE: added step 4 to reproduction recipe
### Describe the bug
The is_incremental() macro always returns false when the target is BigQuery, effectively disabling the incremental build feature.

### Steps To Reproduce
1) Create a fresh DBT project with dbt init
2) Configure the profile to use BigQuery
3) Change the "my_first_dbt_model.sql" file to this:

{{ config(materialized='incremental') }}
select 1 as id
{% if is_incremental() %}
    THIS SHOULD CAUSE AN ERROR
{% endif %}

4) execute "dbt run" to initially create the table
5) execute "dbt run" again
  => no error is displayed (bug)
6) execute "dbt run --bypass-cache"
  => error is displayed (expected behaviour)

### Expected behavior
An error message should be displayed when calling "dbt run" because the SQL inside the is_incremental condition is invalid. 
When running "dbt run --bypass-cache" the result is as expected.

### Additional information
I did a little debugging and verified that is_incremental is indeed returning false. The cause is that adapter.get_relation apparently returns None. Since the problem vanishes when --bypass-cache is specified, I suspect that the problem is caused by a bug in the relation cache.

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [X ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
installed version: 0.14.2
   latest version: 0.14.2

Up to date!
```

**The operating system you're using:**
macOS Mojave

**The output of `python --version`:**
Python 3.7.4

### LINKED ISSUE #1887: Use client.delete_dataset on BigQuery to drop datasets
### Describe the feature
dbt should use the [client.delete_dataset](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.client.Client.html#google.cloud.bigquery.client.Client.delete_dataset) method to atomically drop BigQuery datasets:
```
client.delete_dataset(dataset_id, delete_contents=True, not_found_ok=True)
```

This is an improvement over the existing implementation which first deletes all of the tables in the dataset, then deletes the empty dataset. Some users have reported dbt failing to delete _some_ of the tables in a dataset (and not removing the dataset itself), and I suspect that this change will fix that issue. At the very least, the `client.delete_dataset` implementation is simpler than the [existing implementation](https://github.com/fishtown-analytics/dbt/blob/f985902a002fba36f6f709c6aacf9ae20778e58c/plugins/bigquery/dbt/adapters/bigquery/connections.py#L291-L299).

### PR DESCRIPTION
**Pull request by beckjake**
Fixes #1810
Fixes #1887 

Refactor Relations vs InformationSchemas to handle BQ better
Fix a bug where bigquery cached uppercase schema names wrong.
Use `exists_ok`, `not_found_ok`, and `delete_contents` flags on bigquery dataset operations instead of implementing them manually.

### CODE REVIEW

File: plugins/bigquery/dbt/adapters/bigquery/connections.py [hunk 1]
```diff
@@ -294,7 +294,14 @@ def drop_dataset(self, database, schema):
         client = conn.handle
 
         with self.exception_handler('drop dataset'):
-            for table in client.list_tables(dataset):
+            try:
```
Comment (drewbanin) [inline plugins/bigquery/dbt/adapters/bigquery/connections.py]: can we replace this with:
```
client.delete_dataset(dataset_id, delete_contents=True, not_found_ok=True)
```

If the version of `google-cloud-bigquery` that we use supports it? If not, we can tackle it in a subsequent PR / release.

via [the bq docs](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.client.Client.html#google.cloud.bigquery.client.Client.delete_dataset)

Comment (beckjake) [reply plugins/bigquery/dbt/adapters/bigquery/connections.py]: It looks like we can set a minimum of `1.15.0` and be sure we have that flag available - it's hard to find any information about earlier versions.

Comment (drewbanin) [reply plugins/bigquery/dbt/adapters/bigquery/connections.py]: ok - let's make that change in a separate PR, I'll queue up an issue

**Comment by beckjake** [comment]
I had to rebase onto louisa-may-alcott and drop the `setup.py` changes - no actual code changes - because the query comments branch merged and it also had that change.

**Pull request merged**